ext/intl/tests/gh17317.phpt

```diff
@@ -0,0 +1,56 @@
+--TEST--
+GH-17319 (ResourceBundle iterator crash on NULL key)
+--EXTENSIONS--
+intl
+--CREDITS--
+KidFlo
+--FILE--
+<?php
+foreach ((new ResourceBundle('', NULL))->get('calendar')->get('buddhist') as $key => $value) {
+	echo "KEY: "; var_dump($key);
+	echo "VALUE: "; var_dump($value);
+}
+?>
+--EXPECT--
```

Comment:
Should use `--EXPECTF--` here to be able to use placeholders for the specifc filenames (and line numbers) of the warnings below.

And maybe the test is overspecified; I have no idea how likely it is that the details of this resource bundle will change in ICU versions.